File: JAVA/Maximum Index.java
class Solution{
    static int maxIndexDiff(int a[], int n) { 
        int st = 0, end = n-1;
        int diff = 0;
        while(st<=end){
            if(a[st]<=a[end]){
                diff = Math.max(diff,end-st);
                st++;
                end=n-1;
            }else{
                end--;
            }
        }
        return diff;
    }
}

File: C++/Maximum Index.cpp
class Solution{
    public:
    int maxIndexDiff(int a[], int n) 
    { 
        int i=0;
        int j= n-1;
        int maxi = INT_MIN;
        
        while(i<n)
        {
            if(a[i]>a[j])
            {
                j--;
            }
            else{
                maxi = max(maxi,j-i);
                j = n-1;
                i++;
            }
        }
        return maxi;
    }
};
